### PR TITLE
MM-12710 Migration of team_actions to redux

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
Migration of these actions.

1. removeUserFromTeam(Updated)
2. addUserToTeamFromInvite(Updated)
3. addUsersToTeam(Updated)
4. getInviteInfo(Removed using a redux action instead)
5. switchTeams(Updated)
6. inviteMembers(Removed using a redux action instead)

QA Areas:
1. Removing user from team
2. Adding user to team
3. User signing up with invite team link
4. Sending email invites to users

#### Ticket Link
[MM-12710](https://mattermost.atlassian.net/browse/MM-12710)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has redux changes (https://github.com/mattermost/mattermost-redux/pull/683) Merged

### CODE REVIEW

File: components/admin_console/manage_teams_modal/remove_from_team_button.jsx [hunk 1]
```diff
@@ -5,32 +5,30 @@ import PropTypes from 'prop-types';
...
+    handleClick = async (e) => {
         e.preventDefault();
 
-        removeUserFromTeam(
-            this.props.team.id,
-            this.props.user.id,
-            this.handleMemberRemove,
-            this.props.onError
-        );
+        const {data, error} = await this.props.removeUserFromTeam(this.props.team.id, this.props.user.id);
```
Comment (enahum) [inline components/admin_console/manage_teams_modal/remove_from_team_button.jsx]: this function and the one in `manage_teams_dropdown.jsx` basically does the same thing, perhaps you can combine it into one function and pass that as just one prop?

File: components/invite_member_modal/invite_member_modal.jsx [hunk 2]
```diff
@@ -55,6 +54,9 @@ class InviteMemberModal extends React.PureComponent {
         defaultChannelName: PropTypes.string.isRequired,
         teamType: PropTypes.string.isRequired,
         onHide: PropTypes.func.isRequired,
+        actions: PropTypes.shape({
+            sendEmailInvitesToTeam: PropTypes.func.isRequired,
+        }),
```
Comment (enahum) [inline components/invite_member_modal/invite_member_modal.jsx]: nit: mark the actions as required

File: components/select_team/select_team.jsx [hunk 3]
```diff
@@ -63,20 +63,18 @@ export default class SelectTeam extends React.Component {
...
-                this.props.history.push(`/${team.name}/channels/town-square`);
-            },
-            (error) => {
-                this.setState({
-                    error,
-                    loadingTeamId: '',
-                });
-            }
-        );
+        const {data, error} = await this.props.actions.addUserToTeamFromInvite('', team.invite_id);
```
Comment (enahum) [inline components/select_team/select_team.jsx]: what is the first parameter `token` used for I can't remember...

looking at the api doc it seems inconsistent as it asks for a token but there is nothing about the invite_id https://api.mattermost.com/#tag/teams%2Fpaths%2F~1teams~1%7Bteam_id%7D~1members%2Fpost 

@jwilander thoughts?

File: components/invite_member_modal/invite_member_modal.jsx (hunk 2 above)
Comment (sudheerDev) [reply components/invite_member_modal/invite_member_modal.jsx]: It is marked as required.

File: components/admin_console/manage_teams_modal/remove_from_team_button.jsx (hunk 1 above)
Comment (sudheerDev) [reply components/admin_console/manage_teams_modal/remove_from_team_button.jsx]: I don't follow what you meant as passing them as one prop here. Could you help out with some pseudo logic here? 

`handleMemberRemove` is a prop and it has some state associated in the parent so we need a callback on success. same case with onError

File: components/invite_member_modal/invite_member_modal.jsx (hunk 2 above)
Comment (enahum) [reply components/invite_member_modal/invite_member_modal.jsx]: what I mean is
```js
actions: PropTypes.shape({
    sendEmailInvitesToTeam: PropTypes.func.isRequried,
}).isRequired,
```

File: components/admin_console/manage_teams_modal/remove_from_team_button.jsx (hunk 1 above)
Comment (enahum) [reply components/admin_console/manage_teams_modal/remove_from_team_button.jsx]: 0/5 here but the parent component could pass a prop like `handleRemoveUserFromTeam` then in the parent component you have

```js
async handleRemoveUserFromTeam() {
    const {actions, user, team} = this.props;

    const {data, error} = actions.removeUserFromTeam(team.id, user.id);
     if (data) {
       this.handleMemberRemove();
    } else if (error) {
       this.handleError(error.message);
    }
}
```

then for both components you should only need a prop for `handleRemoveUserFromTeam` and you can remove the props for `removeUserFromTeam`, `onMemberRemove` and `onError`

File: components/select_team/select_team.jsx (hunk 3 above)
Comment (jwilander) [reply components/select_team/select_team.jsx]: The documentation is just missing the invite ID, you can use that as well

File: components/admin_console/manage_teams_modal/remove_from_team_button.jsx (hunk 1 above)
Comment (sudheerDev) [reply components/admin_console/manage_teams_modal/remove_from_team_button.jsx]: Nice call. Made changes accordingly.

**Comment by DHaussermann** [comment]
@sudheerDev I have done some testing as you've outlined in the summary and I may have found an issue here.
When trying to send out a team invite email I get message `Unable to find the existing team`

Steps: 
- Select a team
- Select hamburger menu>> Send email invite 
- Add address and name 
- Click Send invite

emailinvite
 

It's possible this may be server configuration on the SpinMint I enabled the setting and setup SMTP normally - Can you take a look and see if you can repo this.

**Comment by DHaussermann** [comment]
@sudheerDev I was unable to test the latest round. The server would not build because a re-base was  required. Can you assist?

**Comment by DHaussermann** [comment]
Thnaks @sudheerDev  this is now resolved. Also re-tested other team related actions.

This looks good to me.

**Pull request merged**